gentoopm/basepm/upstream.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/python
+# 	vim:fileencoding=utf-8
+# (c) 2024 Anna <cyber@sysrq.in>
+# Released under the terms of the 2-clause BSD license.
+
+from dataclasses import dataclass
+from enum import Enum, auto
+
+
+class PMUpstreamMaintainerStatus(Enum):
+    """ Maintainer status enumeration. """
+
+    NONE = auto()
+    ACTIVE = auto()
+    INACTIVE = auto()
+
+
+@dataclass(frozen=True, order=True)
+class PMUpstreamMaintainer:
+    """ Representation of an upstream maintainer"""
+
+    name: str
+    email: str | None = None
```

Comment:
```suggestion
    email: typing.Optional[str] = None
```

for consistency with existing codebase.